src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -226,13 +225,49 @@ async def _expand_interpreter_search_paths(
     return _SearchPaths(tuple(expanded))
 
 
+# This method is copied from the pex package, located at pex.variables.Variables._get_kv().
+# It is copied here to avoid a hard dependency on pex.
+def _get_kv(variable: str) -> Optional[list[str]]:
+    kv = variable.strip().split("=")
+    if len(list(filter(None, kv))) == 2:
+        return kv
+    else:
+        return None
+
+
+# This method is copied from the pex package, located at pex.variables.Variables.from_rc().
+# It is copied here to avoid a hard dependency on pex.
+def _read_pex_rc(rc: Optional[str] = None) -> dict[str, str]:
+    """Read pex runtime configuration variables from a pexrc file.
+
+    :param rc: an absolute path to a pexrc file.
+    :return: A dict of key value pairs found in processed pexrc files.
+    """
+    ret_vars = {}
+    rc_locations = [
+        os.path.join(os.sep, "etc", "pexrc"),
+        os.path.join("~", ".pexrc"),
+        os.path.join(os.path.dirname(sys.argv[0]), ".pexrc"),
+    ]
+    if rc:
+        rc_locations.append(rc)
+    for filename in rc_locations:
+        try:
+            with open(os.path.expanduser(filename)) as fh:
+                rc_items = map(_get_kv, fh)
+                ret_vars.update(dict(filter(None, rc_items)))
+        except OSError:
+            continue
+    return ret_vars
```

Comment:
This involves reading the pexrc file, which kind of explains why we were directly importing from pex before. I'm not sure if it's worth keeping a hard dependency just for this, but maybe it is in case that logic changes in the future.